platform/darwin/src/MGLStyle.mm

```diff
@@ -127,17 +127,17 @@ - (NSString *)name {
 
 #pragma mark Sources
 
-- (NS_MUTABLE_SET_OF(MGLSource *) *)sources {
+- (NS_SET_OF(__kindof MGLSource *) *)sources {
     auto rawSources = self.mapView.mbglMap->getSources();
-    NSMutableSet *sources = [NSMutableSet setWithCapacity:rawSources.size()];
+    NS_MUTABLE_SET_OF(__kindof MGLSource *) *sources = [NSMutableSet setWithCapacity:rawSources.size()];
     for (auto rawSource = rawSources.begin(); rawSource != rawSources.end(); ++rawSource) {
         MGLSource *source = [self sourceFromMBGLSource:*rawSource];
         [sources addObject:source];
     }
     return sources;
```

Comment:
I’d prefer to return the mutable array here instead of incurring the potential cost of a copy, because the variable that holds the array has local scope and the declared return type is immutable. [It would be a programming error](http://stackoverflow.com/a/27392471/4585461) for the caller to cast the object to an NSMutableArray, even conditionally.